autoPyTorch/pipeline/components/preprocessing/tabular_preprocessing/feature_preprocessing/FeatureAgglomeration.py

```diff
@@ -0,0 +1,122 @@
+        self.pooling_func_mapping: Dict[str, Callable] = dict(mean=np.mean,
+                                                              median=np.median,
+                                                              max=np.max)
+
+        super().__init__(random_state=random_state)
+
+    def fit(self, X: Dict[str, Any], y: Any = None) -> BaseEstimator:
+        if not callable(self.pooling_func):
+            self.pooling_func = self.pooling_func_mapping[self.pooling_func]
+
+        self.preprocessor['numerical'] = SklearnFeatureAgglomeration(
+            n_clusters=self.n_clusters, affinity=self.affinity,
+            linkage=self.linkage, pooling_func=self.pooling_func)
+
+        return self
+
+    @staticmethod
+    def get_hyperparameter_search_space(
+        dataset_properties: Optional[Dict[str, BaseDatasetPropertiesType]] = None,
+        n_clusters: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter='n_clusters',
+                                                                          value_range=(0.5, 0.9),
+                                                                          default_value=0.5,
+                                                                          ),
+        affinity: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter='affinity',
+                                                                        value_range=("euclidean",
+                                                                                     "manhattan",
+                                                                                     "cosine"),
+                                                                        default_value="euclidean",
+                                                                        ),
+        linkage: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter='linkage',
+                                                                       value_range=("ward", "complete", "average"),
+                                                                       default_value="ward",
+                                                                       ),
+        pooling_func: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter='pooling_func',
+                                                                            value_range=("mean", "median", "max"),
+                                                                            default_value="max",
+                                                                            ),
+    ) -> ConfigurationSpace:
+        if dataset_properties is not None:
+            n_features = len(dataset_properties['numerical_columns']) if isinstance(
+                dataset_properties['numerical_columns'], List) else 0
+            if n_features == 1:
+                log = False
+            else:
+                log = n_clusters.log
```

Comment:
Why False when `n_features == 1`?

Reply:
because in that case the range of n_clusters is 0,1 and config space raises an error `ValueError: Negative lower bound (0) for log-scale hyperparameter a is forbidden.`